app/src/main/java/net/sourceforge/opencamera/preview/Preview.java

```diff
@@ -5765,6 +5722,104 @@ protected void onPostExecute(Boolean success) {
                 this.showToast(null, R.string.video_no_free_space);
             }
         }
+
+        return successful;
+    }
+
+    /**
+     * Start video recording.
+     *
+     * {@link #prepareVideoRecording} is meant to be called with the provided {@link MediaRecorder}
+     * and {@link VideoProfile} before calling this method.
+     */
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    private void startVideoRecording(final boolean max_filesize_restart, final MediaRecorder local_video_recorder, final VideoProfile profile) {
+        if (MyDebug.LOG) {
+            Log.d(TAG, "startVideoRecording");
+            Log.d(TAG, "about to start video recorder");
+        }
+
+        applicationInterface.cameraInOperation(true, true);
+        applicationInterface.startingVideo();
+
+        if( mVideoFrameInfoWriter != null ) {
+            Log.d(TAG, "preparing videoFrameInfoWriter");
+
+            try {
+                mVideoFrameInfoWriter.prepare();
+            } catch (IOException e) {
+                Log.e(TAG, "Exception preparing video frame info recorder");
+                e.printStackTrace();
+                applicationInterface.onFrameInfoRecordingFailed();
+                applicationInterface.stoppingVideo();
+
+                video_recorder.reset();
+                video_recorder.release();
+                video_recorder = null;
+                video_recorder_is_paused = false;
+                applicationInterface.cameraInOperation(false, true);
+                this.reconnectCamera(true);
+            }
+        }
+
+        try {
+            local_video_recorder.start();
+            if( test_video_failure ) {
+                if( MyDebug.LOG )
+                    Log.d(TAG, "test_video_failure is true");
+                throw new RuntimeException();
+            }
+            this.video_recorder = local_video_recorder;
+            videoRecordingStarted(max_filesize_restart);
+        }
+        catch(RuntimeException e) {
+            // needed for emulator at least - although MediaRecorder not meant to work with emulator, it's good to fail gracefully
+            Log.e(TAG, "runtime exception starting video recorder");
+            e.printStackTrace();
+            this.video_recorder = local_video_recorder; // still assign, so failedToStartVideoRecorder() will release the video_recorder
+            // told_app_starting must be true if we're here
+            applicationInterface.stoppingVideo();
+            failedToStartVideoRecorder(profile);
+        }
+
+        /*final MediaRecorder local_video_recorder_f = local_video_recorder;
```

Comment:
A lot of commented code in this file --- why we still need it?

Reply:
This huge piece was actually left there by the original developers so I left it untouched too. I will remove the little pieces above created by me as they are not needed anymore, but I'm not sure what to do with this one.

Reply:
This huge piece was actually left there by the original developers so I left it untouched too. I will remove the little pieces above created by me as they are not needed anymore, but I'm not sure what to do with this one.